port_ocean/config/base.py

```diff
@@ -53,15 +53,27 @@ def load_from_config_provider(config_provider: str) -> Any:
 
 
 def parse_providers(
+    settings_model: BaseModel | ModelMetaclass,
     config: dict[str, Any],
     existing_data: dict[str, Any],
 ) -> dict[str, Any]:
     """
     Normalizing the config yaml file to work with snake_case and getting only the data that is missing for the settings
     """
     for key, value in config.items():
-        if isinstance(value, dict):
-            existing_data[key] = parse_providers(value, existing_data.get(key, {}))
+        if isinstance(value, dict) and settings_model is not None:
+            # If the value is ModelMetaClass typed then its a nested model, and we need to parse it
+            # If the value is a primitive dict then we need to decamelize the keys and not recurse into the values because its no longer part of the model
```

Comment:
```suggestion
            # If the value is of type ModelMetaClass then its a nested model, and we need to parse it
            # If the value is of type primitive dict then we need to decamelize the keys and not recurse into the values because its no longer part of the model
```